osu.Game/Rulesets/UI/Scrolling/DrawableScrollingRuleset.cs

```diff
@@ -78,7 +78,7 @@ public abstract class DrawableScrollingRuleset<TObject> : DrawableRuleset<TObjec
         /// <summary>
         /// The <see cref="MultiplierControlPoint"/>s that adjust the scrolling rate of <see cref="HitObject"/>s inside this <see cref="DrawableRuleset{TObject}"/>.
         /// </summary>
-        protected readonly SortedList<MultiplierControlPoint> ControlPoints = new SortedList<MultiplierControlPoint>(Comparer<MultiplierControlPoint>.Default);
+        public readonly MultiplierControlPointList ControlPoints = new MultiplierControlPointList(Comparer<MultiplierControlPoint>.Default);
```

Comment:
This can't be public in its current state. If it is required for some reason or another I think it needs to be in its own isolated PR.

This PR should *only* touch the `TaikoModHidden` file, nothing else.

Reply:
To rephrase before you add all these changes back: I don't think any of this should be required in the first place, and this mod should be using `IApplicableToBeatmap` and then adjusting via `ControlPointInfo`.

Reply:
That would require a lot of code duplication, is that alright?

Reply:
Which code and how much?

Reply:
all the code to generate MultiplierControlPoint's from ControlPointInfo

Reply:
I think it will be easier to write this from scratch myself rather than attempt to work with this PR. See how you go, I think it should be possible. I strongly believe there's no need to consume the post-processed control points.

Reply:
you were right, a lot of MultiplierControlPoint's complexity is coming from mania